runtime/src/identity.rs

```diff
@@ -135,6 +137,9 @@ decl_storage! {
 
         /// Nonce to ensure unique DIDs are generated. starts from 1.
         pub DidNonce get(did_nonce) build(|_| 1u128): u128;
+
+        /// Pre-authorize join to Identity.
+        pub PreAuthorizedJoinDid get( pre_authorized_join_did): map Key => Vec<PreAuthorizedKeyInfo>;
```

Comment:
Let's use `map (Key, DID) => PreAuthorizedKeyInfo` instead of `map Key => Vec<PreAuthorizedKeyInfo>`. I don't think we need to create a vector here.

Reply:
From slack chat, the proposed changes are:

1. Use `DobuleMap` instead of `map` of `vec`. 
2. Replace "automatic clear" of pre-auths once key accepts one DID, with a manual process. Master key of the DID and the pre-auth signing key will be able to revoke the invitation.